enigma-core/enclave/src/lib.rs

```diff
@@ -266,12 +266,22 @@ fn decrypt_inputs(callable: &[u8], args: &[u8], inputs_key: &DhKey) -> Result<(V
     Ok((decrypted_args, function_name))
 }
 
-fn save_enc_delta(db_ptr: *const RawPointer, delta: &Option<EncryptedPatch>) -> Result<Hash256, EnclaveError> {
+fn get_enc_delta(delta: &Option<EncryptedPatch>) -> Hash256 {
     if let Some(delta) = delta {
-        enigma_runtime_t::ocalls_t::save_delta(db_ptr, delta)?;
-        Ok(delta.keccak256_patch())
+        delta.keccak256_patch()
     } else {
-        Ok(Hash256::default())
+        Hash256::default()
+    }
+}
+
+fn store_delta_and_state(db_ptr: *const RawPointer, delta: &Option<EncryptedPatch>, state: &ContractState) -> Result<(), EnclaveError> {
+    match delta {
+        Some(d) => {
+            enigma_runtime_t::ocalls_t::save_delta(db_ptr, d)?;
```

Comment:
Here I would suggest to minimize the probability of error between saving delta and state and move encryption of the state before saving delta.

Reply:
done